### DISCUSSION TITLE
Deleting files via trees no longer works?

### DISCUSSION THREAD

**Discussion by chrisdothtml** [community/community#24420] [Apps, API and Webhooks]
<p>Based on <a href="https://stackoverflow.com/a/23714505/2880820" rel="nofollow noopener">this comment</a>, I’ve been able to delete multiple files from a repo with the v3 api via:</p>
<ol>
<li>
<p>GET the parent tree recursively</p>
</li>
<li>
<p>Remove the files I want to delete from the tree</p>
</li>
<li>
<p>POST a new tree, using the parent tree sha as the <code>base_tree</code></p>
</li>
</ol>
<p>This used to work fine, but no longer does. It currently responds with “201 Created”, but just contains the parent tree (same sha, and still containing deleted files). Have there been any recent changes to the v3 api that may have caused this?</p>
<p>EDIT: this also breaks renaming files via trees</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>I just tested it and it worked fine. Here’s what I did:</p>
<ol>
<li>Create new directory <code>directory</code> in <a href="https://github.com/lee-dohm/test-repo" rel="nofollow noopener">https://github.com/lee-dohm/test-repo</a> with two files <code>delete-this-file.txt</code> and <code>do-not-delete-this-file.txt</code> (<a href="https://github.com/lee-dohm/test-repo/tree/c3c0e292a3bd10117286d790cb2ddd65a7eb30c7" rel="nofollow noopener">see result</a>)</li>
<li>Create a new tree object for <code>directory</code> that does not contain <code>delete-this-file.txt</code>
</li>
<li>Create a new tree object for <code>/</code> that points to all the same stuff except points to the new tree object created in step 2 for <code>directory</code>
</li>
<li>Create a commit object pointing to the tree object created in step 3 and the parent commit created in step 1</li>
<li>Update the master branch to point to the new commit object created in step 4</li>
</ol>
<p>You can see the <a href="https://github.com/lee-dohm/test-repo/tree/3677c2e495bb9d30ebe61d5cba6f4a675a743c48" rel="nofollow noopener">final result</a> in the repository.</p>
<p>One thing I did notice is that when creating the new tree objects, I needed to leave out the <code>base_tree</code> value.</p>
<p>I hope that helps!</p>

**Comment by chrisdothtml** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> thanks for the response!</p>
<blockquote>
<p>One thing I did notice is that when creating the new tree objects, I needed to leave out the <code>base_tree</code> value.</p>
</blockquote>
<p>This was the part I had wrong. I was passing the parent commit’s tree sha, but seems like doing so will fill in missing leafs from the base tree. My script definitely previously worked (improperly) with<code>base_tree</code>, so maybe there was a recent update that fixed it?</p>
<p>Either way, leaving it out fixed my issue.</p>
<p>Thanks!</p>
